sys/mips/malta/malta_machdep.c

```diff
@@ -342,22 +346,87 @@ malta_cpu_freq(void)
 	return (platform_counter_freq);
 }
 
+#ifdef __CHERI_PURE_CAPABILITY__
+static void
+platform_clear_bss(void *kroot)
+{
+	/*
+	 * We need to hand-craft a capability for edata because it is defined
+	 * by the linker script and have no size info.
+	 */
+	void *edata_start;
+	size_t edata_siz = (__cheri_addr size_t)(&end) -
+	    (__cheri_addr size_t)(&edata);
+
+	edata_start = cheri_ptrperm(
+	    cheri_setaddress(kroot, (__cheri_addr vaddr_t)(&edata)),
+	    edata_siz, CHERI_PERM_STORE);
+	memset(edata_start, 0, edata_siz);
+}
+
+static char *
+platform_argv_ptr(int32_t argv)
+{
+	char *arg;
+	size_t len;
+
+	/*
+	 * YAMON uses 32bit pointers to strings so
+	 * sign-extend them to the correct type manually.
+	 */
+	arg = cheri_andperm(
+	    cheri_setaddress(mips_kseg0_cap, (vm_offset_t)(argv)),
+	    CHERI_PERM_LOAD | CHERI_PERM_STORE);
+	arg = cheri_setboundsexact(arg, strlen(arg) + 1);
```

Comment:
This seems like a bad idea

Reply:
Elaborate. I think we already need to trust that the strings given to use are well initialized, the alternative would be to not bound the capability (which is essentially the same as trusting the zero-terminator) or to set an arbitrary maximum length (but then we can not guarantee it will not overlap with anything else, I think). Ideally the loader would give an argv populated with capabilities.

Reply:
As in it shouldn't be exact, since that'll go pear-shaped for (stupidly) long strings that are technically valid

Reply:
Makes sense, I'll drop it.

Reply:
Still using cheri_setboundsexact rather than cheri_setbounds